core/parser.mly

```diff
@@ -443,6 +448,12 @@ tlfunbinding:
 | OP OPERATOR pattern perhaps_location block                   { ((dl_unl, false), $2, [[$3]], $4, $5)          }
 | OP pattern OPERATOR perhaps_location block                   { ((dl_unl, false), $3, [[$2]], $4, $5)          }
 
+switch_tlfunbinding:
+| fun_kind VARIABLE arg_lists perhaps_location switch_body     { ($1, $2, (switch_fun_currying_guard $loc($3) $3), $4, $5)   }
```

Comment:
Note an alternative to the dynamic check `switch_fun_currying_guard`, would be to restrict the grammar such that only uncurried function definitions are admissible -- amounts to a static enforcement of `switch_fun_currying_guard`.

I am not advocating you should restrict the grammar as we ultimately want to support curried function definitions, I just wanted to make sure you were aware of this alternative design decision.

Reply:
I have tried changing grammar before - menhir gave me errors of `one state has shift/reduce conflicts.` I think its because the new grammar is not LL(1). So I just use dynamic checking.

Reply:
Menhir is a LR(1) parser generator. Note that `LL != LR`.

There are typically ways in which one can "massage" the grammar to eliminate shift/reduce conflicts. It ultimately depends on how you factor the production rules.